app/controllers/api/container_deployments_controller.rb

```diff
@@ -9,7 +9,12 @@ def options
       # TODO: this service is rendering resources which (a) require
       # authentication (problematic for CORS compatibility), and (b)
       # are not being properly filtered by RBAC
-      render_options(:container_deployments, ContainerDeploymentService.new.all_data)
+      if [HttpHeaders::MIQ_TOKEN, HttpHeaders::AUTH_TOKEN, "HTTP_AUTHORIZATION"].any? { |header| request.headers.include?(header) }
+        require_api_user_or_token
+        render_options(:container_deployments, ContainerDeploymentService.new.all_data)
```

Comment:
We _could_, although personally I think this is an anti-pattern and I don't want to promote re-use of it. I could extract a private method in this controller if you think this would be better?